framework/web/UrlManager.php

```diff
@@ -428,6 +429,24 @@ public function createUrl($params)
     }
 
     /**
+     * Returns the value indicating whether result of [[createUrl()]] of rule should be cached in internal cache.
+     *
+     * @param UrlRuleInterface $rule
+     * @return bool|null `true` if result should be cached, `false` if not, `null` if rule does not provide info about
```

Comment:
On second thought we could treat `null` as `true` - such rules will take precedence (as they're always cached) over regular rules, but this still should be more efficient than disabling the whole cache because of one rule which not implements `getCreateUrlStatus()`.